x/protorev/keeper/epoch_hook_test.go

```diff
@@ -0,0 +1,74 @@
+package keeper_test
+
+import (
+	"fmt"
+
+	"github.com/osmosis-labs/osmosis/v13/x/protorev/types"
+)
+
+func (suite *KeeperTestSuite) TestEpochHook() {
+	// All of the pools initialized in the setup function are available in keeper_test.go
+	// akash <-> types.OsmosisDenomination
+	// juno <-> types.OsmosisDenomination
+	// ethereum <-> types.OsmosisDenomination
+	// bitcoin <-> types.OsmosisDenomination
+	// canto <-> types.OsmosisDenomination
+	// and so on...
+	expectedToSee := make(map[string]Pool)
+	for _, pool := range suite.pools {
+
+		// Module currently limited to two asset pools
+		// Instantiate asset and amounts for the pool
+		if len(pool.PoolAssets) == 2 {
+			pool.Asset1 = pool.PoolAssets[0].Token.Denom
+			pool.Amount1 = pool.PoolAssets[0].Token.Amount
+			pool.Asset2 = pool.PoolAssets[1].Token.Denom
+			pool.Amount2 = pool.PoolAssets[1].Token.Amount
+		}
+
+		if pool.Asset1 == types.OsmosisDenomination || pool.Asset2 == types.OsmosisDenomination || pool.Asset1 == types.AtomDenomination || pool.Asset2 == types.AtomDenomination {
```

Comment:
Shouldn't this be `Asset1 == osmo || Asset1 == atom && Asset2 == osmo || Asset2 == atom` I thought the `tokenInDenom` **and** `tokenOutDenom` had to be either osmo or atom

Reply:
tldr the pools that we instantiate in `keeper_test.go` don't always contain either atom or osmo because we can sandwich a pool that has neither atom or osmo if we pair it with atom or osmo on either side. i.e. if we have pool juno/akash, we can make a route osmo/juno -> juno/akash -> akash/osmo (replace osmo with atom and it is also valid).

`TestEpochHook` is checking the pools that were created in`keeper_test.go` and seeing if any of the assets in the pool are atom or osmo. Once we have all the pools that contain atom, osmo or both, we get the pools that have the highest liquidity for every unique pair. The pools that end up being filtered here should be the same ones that the epoch sets in state.

Reply:
I'll add more documentation on the tests so that its easier to read since `keeper_test.go` is a heavy read (will also look into breaking it up).

Reply:
sounds good